src/controllers/teamCtrl.js

```diff
@@ -216,10 +263,24 @@ exports.getUserOfTeam = (req, res) => {
   Team.findById(req.params.teamId)
     .populate(['users.user'])
     .then((team) => {
+      if (team) {
+        return res
+          .status(404)
+          .json({ message: 'No user found for this team!' });
+      }
+
+      // Blocks users who want to access other users without being admin
+      const isUserInTeam = team.users.some(
+        (user) => user.user.toString() === req.userId
```

Comment:
```suggestion
        (user) => user.user.id.toString() === req.userId
```
'user.id' you mean ?

Reply:
Yes it should be with 'id' but I have to test it before make a change, thanks for the catch !

Reply:
Nice catch! It's write ✅